cfgrammar/src/lib/yacc/parser.rs

```diff
@@ -1332,36 +1285,37 @@ x"
 
     #[test]
     fn test_bad_prec_overrides() {
+        let src = "
+        %%
+        S: 'A' %prec ;
+        ";
         match parse(
             YaccKind::Original(YaccOriginalActionKind::GenericParseTree),
-            "
-          %%
-          S: 'A' %prec ;
-          ",
+            src,
         ) {
             Ok(_) => panic!("Incorrect %prec parsed"),
             Err(YaccParserError {
                 kind: YaccParserErrorKind::IllegalString,
-                line: 3,
+                span,
                 ..
-            }) => (),
+            }) => assert_eq!(line_of_offset(src, span.start()), 3),
             Err(e) => panic!("Incorrect error returned {}", e),
         }
```

Comment:
It might be a good idea to macroize these blocks so we can do the `let src` inside the macro,
since there are a few blocks with multiple `let src` shadowing.  Once we bind it like this there is potential to lookup newlines in the wrong `src` variable.

Reply:
I think there is probably too much magic to stuff in a macro, it may shorten the amount of repetition it but it's probably clearer without, so going to just resolve this.